src/System.Management.Automation/engine/hostifaces/PSTask.cs

```diff
@@ -449,6 +449,10 @@ public void Start(Runspace runspace)
                     Runspace.DefaultRunspace = runspace;
                     runspace.ExecutionContext.SessionState.Internal.SetLocation(_currentLocationPath);
                 }
+                catch (DriveNotFoundException)
+                {
+                    // Allow task to run if current drive is not available.
```

Comment:
Should we emit a warning?

Reply:
Кather, we just need to document that we are using clean space because there may be other side effects for this reason.

Reply:
I thought about it and decided "no", because an appropriate error will occur if/when a path cannot be resolved.  I see this as a partial fix, and the real solution is to provide a new opt-in feature to replicate runspace state.  This PR just addresses the terminating error issue.

Reply:
In this case, what directory is the script running in?  $PSHOME?

Reply:
I guess we use default came from .Net and OS.

Reply:
Whatever the default PowerShell location is.  For Linux I believe it is $HOME.

Reply:
I guess it is here:
https://github.com/PowerShell/PowerShell/blob/238cb5c8dcc01565d0b93ade2faabfd8e6563329/src/System.Management.Automation/engine/InitialSessionState.cs#L3362-L3374

Reply:
Ok, seems fine.